### DISCUSSION TITLE
Creation of users/application that are not localhost

### DISCUSSION THREAD

**Discussion by Gupta-Krishnaa** [elsa-workflows/elsa-documentation#15] [Q&A]
In the Elsa Documentation for 3.0, in order to create a new user we are supposed to call the following rest endpoint

```
https://server-name/elsa/api/identity/applications \
--header 'Content-Type: application/json' \
--data '{
    "name": "Postman",
    "roles": ["admin"]
}
```

However, it's applicable only for `localhost` as the [documentation](https://v3.elsaworkflows.io/docs/installation/aspnet-apps-workflow-server) states 

> By default, creating applications is allowed for anonymous requests from the localhost.

I would like to host the application on my personal azure platform with App Service which will not have `localhost`. What should be done in that case ?

`cc`: @sfmskywalker @agriffard @VahidNaderi @mohdali

**Comment by Gupta-Krishnaa** [comment]
Any updates ?

`cc`: @sfmskywalker @agriffard @VahidNaderi @mohdali

**Reply by mohdali** [reply]
I'm not exactly sure but can't you do something from the KUDU console using PowerShell?

**Reply by Gupta-Krishnaa** [reply]
Can you elaborate on this ? Do you mean to say that I should fire a rest call from KUDU powershell which creates a user/application ?

For example:
```
https://server-name/elsa/api/identity/applications
```

**Reply by mohdali** [reply]
I haven't tried myself but I assume it should work if it acts as a local shell. You can fire a web request from there to see if it creates the desired user.

**Reply by Gupta-Krishnaa** [reply]
I tried as you suggested but I am getting `Forbidden 403`:

Here s the detail:

![image](https://github.com/elsa-workflows/elsa-documentation/assets/132350939/698ed12b-5f62-4c53-9fdc-dde976d93dcf)

![image](https://github.com/elsa-workflows/elsa-documentation/assets/132350939/ab28fbbb-9440-4915-807c-6331ce68362f)

![image](https://github.com/elsa-workflows/elsa-documentation/assets/132350939/242eb8c7-1411-4eed-893f-c08e6935f676)

And Lastly,

![image](https://github.com/elsa-workflows/elsa-documentation/assets/132350939/d046aaab-d913-429e-a3d4-ec260bf7f875)


The result:

![image](https://github.com/elsa-workflows/elsa-documentation/assets/132350939/dbc44e94-ddbe-4897-8a43-334aa36e277c)


Any ideas ?

**Comment by sfmskywalker** [comment]
In an ideal scenario, Elsa Studio would guide users through a setup procedure during its initial launch. This would facilitate the creation of an admin user tailored to the target environment, ensuring a seamless experience.

Currently, Elsa Studio does not possess this feature. As a workaround, I recommend creating the user in a local environment. Once the user is successfully created, you can then migrate the database or the specific database records to your production environment. This method ensures that you have the necessary user configurations in place before deploying to a non-localhost environment like Azure App Service.

It's essential to ensure the security and integrity of your data during this process. Always backup your production database before making any changes and test the migration in a staging environment if possible.
